File: server/lib/pricing.js
import { getSettingNumber } from "./settings.js";
export async function calcFare(distanceKm){
  const base=await getSettingNumber("base_fare",3.50);
  const included=await getSettingNumber("included_km",3);
  const perAfter=await getSettingNumber("per_km_after",0.40);
  const percent=await getSettingNumber("commission_percent",10);
  let fare=base;
  if(distanceKm>included) fare += (distanceKm-included)*perAfter;
  fare=Math.round(fare*100)/100;
  const commission=Math.round((fare*(percent/100))*100)/100;
  return {fare,commission};
}

File: app/services/pricing.py
from decimal import Decimal

DEFAULT_BASE_FARE = Decimal("3.50")
DEFAULT_INCLUDED_KM = Decimal("3.0")
DEFAULT_PER_KM_AFTER = Decimal("0.40")
DEFAULT_COMMISSION_RATE = Decimal("0.10")  # 10%

def calc_fare(distance_km: Decimal,
              base_fare: Decimal = DEFAULT_BASE_FARE,
              included_km: Decimal = DEFAULT_INCLUDED_KM,
              per_km_after: Decimal = DEFAULT_PER_KM_AFTER) -> Decimal:
    if distance_km <= included_km:
        return base_fare
    extra = distance_km - included_km
    return (base_fare + extra * per_km_after).quantize(Decimal("0.01"))

def calc_commission(fare: Decimal, rate: Decimal = DEFAULT_COMMISSION_RATE) -> Decimal:
    return (fare * rate).quantize(Decimal("0.01"))
